compendium/ConfigurationAdmin/src/ConfigurationAdminImpl.cpp

```diff
@@ -462,6 +507,22 @@ namespace cppmicroservices {
             }
           }
         }
+        //TODO - factoryPid
+        const auto configurationListeners =
+          configListenerTracker->GetServices();
+        auto type =
+          removed
+            ? cppmicroservices::service::cm::ConfigurationEventType::CM_REMOVED
+            : cppmicroservices::service::cm::ConfigurationEventType::CM_UPDATED;
+
+        for (auto it : configurationListeners) {
+          notifyListener(
+            pid,
+            "",
+            type,
+            cmContext.GetServiceReference("ConfigurationAdminImpl"),
```

Comment:
Use templatised version of GetServiceReference for ConfigurationAdminImpl instead of string version. Compiler more able to find errors.